src/gcp/cloudsql/permissions_setup.ts

```diff
@@ -0,0 +1,384 @@
+import { Options } from "../../options";
+import {
+  firebaseowner,
+  firebasewriter,
+  firebasereader,
+  ownerRolePermissions,
+  writerRolePermissions,
+  readerRolePermissions,
+  defaultPermissions,
+  FIREBASE_SUPER_USER,
+} from "./permissions";
+import { iamUserIsCSQLAdmin } from "./cloudsqladmin";
+import { setupIAMUsers } from "./connect";
+import { logger } from "../../logger";
+import { confirm } from "../../prompt";
+import * as clc from "colorette";
+import { FirebaseError } from "../../error";
+import { needProjectNumber } from "../../projectUtils";
+import { executeSqlCmdsAsIamUser, executeSqlCmdsAsSuperUser, getIAMUser } from "./connect";
+import { concat } from "lodash";
+import { getDataConnectP4SA, toDatabaseUser } from "./connect";
+
+export type TableMetaData = {
```

Comment:
Nit: MetaData -> Metadata - it is one word, not two.

Applies for schemaMetaData too, and the getter functions